src/App.tsx

```diff
@@ -1,223 +1,167 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import {
+  ErrorNotification,
+} from './components/ErrorNotification/ErrorNotification';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Footer } from './components/Footer/Footer';
+import { Header } from './components/Header/Header';
+import { TodosList } from './components/TodoList/TodoList';
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterType';
+import { getTodos, updateTodos } from './api/todos';
+import { Error } from './types/ErrorType';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.All);
+  const [errorNotification, setErrorNotification] = useState<string | null>('');
+  const [title, setTitle] = useState('');
+  const [selectedTodosIds, setSelectedTodosIds] = useState<number[]>([]);
+  const [isAdding, setIsAdding] = useState(false);
+  const [toggle, setToggle] = useState(true);
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter(({ completed }) => {
+      switch (filterType) {
+        case FilterType.Active:
+          return !completed;
+
+        case FilterType.Completed:
+          return completed;
+
+        default:
+          return true;
+      }
+    });
+  }, [filterType, todos]);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    const getTodosFromServer = async (userId: number) => {
+      try {
+        const todosFromServer = await getTodos(userId);
+
+        setTodos(todosFromServer);
```

Comment:
If the new state is computed better use the previous state, you can pass a function to `setTodos`.